src/pg_client.py

```diff
@@ -22,9 +22,9 @@ def __init__(self, db_urls: list[str]):
         ]
 
     @classmethod
-    def exec_query(cls, query: str, engine: Engine) -> DataFrame:
+    def exec_query(cls, query: str, engine: Engine, dtype=None) -> DataFrame:
```

Comment:
Missing Type annotation:
```suggestion
    def exec_query(cls, query: str, engine: Engine, dtype: SOMETHING=None) -> DataFrame:
```

Reply:
Will add a type there.

The change to this function is one of the changes where I am a bit unsure if it is a good approach. The goal is to avoid automatic conversion of everything to pandas types, e.g., "int64" (or "Int64" or "float64" if there are integers and missing values).

Reply:
Ya pandas types are weak AF. I guess you could cast to string in the query and then int when you get it back.